dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/hibernate/HibernateProgramInstanceStore.java

```diff
@@ -242,6 +243,24 @@ public boolean existsIncludingDeleted( String uid )
         return result != null && result > 0;
     }
 
+    @Override
+    public List<String> getUidsIncludingDeleted( List<String> uids )

```

Comment:
Did we think about a situation with a very large number of TEI uids here?

Reply:
From the tracker side at least, no... we are already partitioning the list, so should not be that big. We can add automatic partitioning to some of these services, but lets do it in separate PR.